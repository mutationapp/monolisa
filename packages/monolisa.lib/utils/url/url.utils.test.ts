import {
  removeLeadingSlashes,
  getParentPath,
  buildQuery,
  parseQuery,
  appendQuery,
  removeTrailingSlashes,
  getDomain,
  addLeadingSlash,
  removeQuery,
  getParentPathUntil,
} from './url.utils'

test.each([undefined, '', 'clean', '/path/to/me/', 'relative/'])(
  'Removes leading slash for: %s',
  (value?: string) => {
    expect(removeLeadingSlashes(value)).toMatchSnapshot()
  },
)

test.each([undefined, 'clean', '/parent/me/', 'relative/me'])(
  'Returns parent path for: %s',
  (value?: string) => {
    expect(getParentPath(value)).toMatchSnapshot()
  },
)

test.each([undefined, {}, { f: 1, g: '2', k: false, j: true }])(
  'Builds the query for: %o',
  (object?: object) => {
    expect(buildQuery(object)).toMatchSnapshot()
  },
)

test.each(['', 'monolisa.lab?', 'monolisa.app?foo=bar&bar=foo'])(
  'parseQuery: %s',
  value => {
    expect(parseQuery(value)).toMatchSnapshot()
  },
)

test.each(['', 'monolisa.lab?', 'monolisa.app?foo=bar&bar=foo'])(
  'removeQuery: %s',
  value => {
    expect(removeQuery(value)).toMatchSnapshot()
  },
)

test.each([
  {
    url: '?foo=bar&bar=foo',
    query: {
      baz: 'baz',
    },
  },
])('appendQuery: %o', ({ url, query }) => {
  expect(appendQuery(url, query)).toMatchSnapshot()
})

test.each([undefined, '/', '//', 'nope'])(
  'removeTrailingSlashes: %s',
  value => {
    expect(removeTrailingSlashes(value)).toMatchSnapshot()
  },
)

test.each(['/', '//', 'nope'])('addLeadingSlash: %s', value => {
  expect(addLeadingSlash(value)).toMatchSnapshot()
})

test.each([
  undefined,
  new URL('http://foo.bar'),
  'http://monolisa.app',
  'http://localhost:3000',
])('removeTrailingSlashes: %s', value => {
  expect(getDomain(value)).toMatchSnapshot()
})

test.each([
  { path: 'foo/bar/baz', until: 'bar' },
  { path: 'oops', until: 'foo' },
])('getParentPathUntil: %o', ({ path, until }) => {
  expect(getParentPathUntil(path, until)).toMatchSnapshot()
})
