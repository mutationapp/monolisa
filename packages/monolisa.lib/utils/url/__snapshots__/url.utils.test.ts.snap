// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Builds the query for: { f: 1, g: '2', k: false, j: true } 1`] = `"f=1&g=2&j=true"`;

exports[`Builds the query for: {} 1`] = `""`;

exports[`Builds the query for: undefined 1`] = `undefined`;

exports[`Removes leading slash for:  1`] = `""`;

exports[`Removes leading slash for: /path/to/me/ 1`] = `"path/to/me/"`;

exports[`Removes leading slash for: clean 1`] = `"clean"`;

exports[`Removes leading slash for: relative/ 1`] = `"relative/"`;

exports[`Removes leading slash for: undefined 1`] = `undefined`;

exports[`Returns parent path for: /parent/me/ 1`] = `"/parent"`;

exports[`Returns parent path for: clean 1`] = `""`;

exports[`Returns parent path for: relative/me 1`] = `"relative"`;

exports[`Returns parent path for: undefined 1`] = `undefined`;

exports[`addLeadingSlash: / 1`] = `"/"`;

exports[`addLeadingSlash: // 1`] = `"/"`;

exports[`addLeadingSlash: nope 1`] = `"/nope"`;

exports[`appendQuery: { url: '?foo=bar&bar=foo', query: { baz: 'baz' } } 1`] = `"?foo=bar&bar=foo&baz=baz"`;

exports[`getParentPathUntil: { path: 'foo/bar/baz', until: 'bar' } 1`] = `"foo/bar"`;

exports[`getParentPathUntil: { path: 'oops', until: 'foo' } 1`] = `"oops"`;

exports[`parseQuery:  1`] = `Object {}`;

exports[`parseQuery: monolisa.app?foo=bar&bar=foo 1`] = `
Object {
  "bar": "foo",
  "foo": "bar",
}
`;

exports[`parseQuery: monolisa.lab? 1`] = `Object {}`;

exports[`removeQuery:  1`] = `""`;

exports[`removeQuery: monolisa.app?foo=bar&bar=foo 1`] = `"monolisa.app"`;

exports[`removeQuery: monolisa.lab? 1`] = `"monolisa.lab"`;

exports[`removeTrailingSlashes: / 1`] = `""`;

exports[`removeTrailingSlashes: // 1`] = `""`;

exports[`removeTrailingSlashes: http://foo.bar/ 1`] = `"foo.bar"`;

exports[`removeTrailingSlashes: http://localhost:3000 1`] = `"localhost"`;

exports[`removeTrailingSlashes: http://monolisa.app 1`] = `"monolisa.app"`;

exports[`removeTrailingSlashes: nope 1`] = `"nope"`;

exports[`removeTrailingSlashes: undefined 1`] = `undefined`;

exports[`removeTrailingSlashes: undefined 2`] = `undefined`;
